oak_functions/loader/src/logger.rs

```diff
@@ -0,0 +1,71 @@
+//
+// Copyright 2021 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use chrono::{SecondsFormat, Utc};
+use log::{Level, LevelFilter};
+use std::default::Default;
+
+/// A simple logger that splits logging between writing logs that contain only public, non-sensitive
+/// content and writing logs that could potentially contain sensitive content.
+///
+/// Writing of potentially sensitive content will be ignored unless the "oak-unsafe" feature is
+/// enabled.
+pub(crate) struct Logger {
+    max_level: LevelFilter,
+}
+
+impl Logger {
+    /// Creates a new logger with the specified maximum `LevelFilter`.
+    #[allow(dead_code)]
+    pub(crate) fn new(max_level: LevelFilter) -> Self {
```

Comment:
Should we just stick to `pub` instead of `pub(crate)` throughout? I don't think it is providing any meaningful advantage, and if we ever decide to use this for something else we would have to change them to `pub` anyways.

Reply:
Done.